### PR TITLE
(FACT-2211) Add OSX mountpoints fact

### PR DESCRIPTION
**Pull request by florindragos**

### CODE REVIEW

File: lib/facts/macosx/mointpoints.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+module Facter
+  module Macosx
+    class Mountpoints
+      FACT_NAME = 'mountpoints'
+
+      def call_the_resolver
+        mountpoints = Resolvers::Macosx::Mountpoints.resolve(FACT_NAME.to_sym)
+
+        fact = {}
+        mountpoints.each do |mnt|
```
Comment (IrimieBogdan) [inline lib/facts/macosx/mointpoints.rb]: There should be a nill for mountpoints.
E.g.

return ResolvedFact.new(FACT_NAME, nil) if mountpoints.nil?

File: lib/facts/macosx/mointpoints.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module Facter
+  module Macosx
+    class Mountpoints
+      FACT_NAME = 'mountpoints'
+
+      def call_the_resolver
+        mountpoints = Resolvers::Macosx::Mountpoints.resolve(FACT_NAME.to_sym)
+        return ResolvedFact.new(FACT_NAME, nil) if mountpoints.nil?
```
Comment (oanatmaria) [inline lib/facts/macosx/mointpoints.rb]: unless mountpoints # to avoid nil check

File: spec/facter/facts/macosx/mountpoints_spec.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  context '#call_the_resolver' do
+    it 'returns a fact' do
+      expected_fact = double(Facter::ResolvedFact, name: 'mountpoints', value: parsed_fact)
+      allow(Facter::Resolvers::Macosx::Mountpoints).to receive(:resolve).with(:mountpoints).and_return(resolver_output)
+      allow(Facter::ResolvedFact).to receive(:new).with('mountpoints', parsed_fact).and_return(expected_fact)
+
+      fact = described_class.new
+      expect(fact.call_the_resolver).to eq(expected_fact)
+    end
+  end
```
Comment (oanatmaria) [inline spec/facter/facts/macosx/mountpoints_spec.rb]: Could you  please change the style to something like https://github.com/puppetlabs/facter-ng/pull/243/files#diff-2b4d178225c7b1c48c19f2806f378364

File: lib/resolvers/utils/filesystem_helper.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+module Facter
+  module FilesystemHelper
+    MOUNT_KEYS = %i[device filesystem path options
+                    available available_bytes size
+                    size_bytes used used_bytes capacity].freeze
+
+    def self.read_mountpoints
+      require 'sys/filesystem'
```
Comment (oanatmaria) [inline lib/resolvers/utils/filesystem_helper.rb]: this require should be mocked in : spec/mocks/kernel_mock.rb

**Review by BogdanIrimie — APPROVED** [review]

**Pull request merged**